### PR TITLE
eventfd: Remove the unused and private eventfd_get_minor

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
since it isn't defined by Linux kernel too

## Impact
Remove eventfd_get_minor

## Testing
Pass CI

### CODE REVIEW

File: include/nuttx/fs/ioctl.h [hunk 1]
```diff
@@ -169,10 +169,6 @@
                                            *      int value.
                                            * OUT: Origin option.
                                            */
-#define FIOC_MINOR      _FIOC(0x000c)     /* IN:  None
-                                           * OUT: Integer that contains device
-                                           *      minor number
-                                           */
 #define FIOCLEX         _FIOC(0x000d)     /* IN:  None
```
Comment (pkarashchenko) [inline include/nuttx/fs/ioctl.h:172]:
```suggestion
#define FIOCLEX         _FIOC(0x000c)     /* IN:  None
```

File: fs/vfs/fs_eventfd.c [hunk 2]
```diff
@@ -121,8 +119,8 @@ static const struct file_operations g_eventfd_fops =
   eventfd_do_close, /* close */
   eventfd_do_read,  /* read */
   eventfd_do_write, /* write */
-  0,                /* seek */
-  eventfd_do_ioctl  /* ioctl */
+  NULL,             /* seek */
+  NULL,             /* ioctl */
```
Comment (pkarashchenko) [inline fs/vfs/fs_eventfd.c:123]:
```suggestion
  NULL              /* ioctl */
```

File: include/nuttx/fs/ioctl.h (hunk 1 above)
Comment (pkarashchenko) [inline include/nuttx/fs/ioctl.h:172]:
```suggestion
#define FIOCLEX         _FIOC(0x000c)     /* IN:  None
```

File: fs/vfs/fs_eventfd.c (hunk 2 above)
Comment (xiaoxiang781216) [reply fs/vfs/fs_eventfd.c:123]: Done.

File: include/nuttx/fs/ioctl.h (hunk 1 above)
Comment (xiaoxiang781216) [reply include/nuttx/fs/ioctl.h]: Done.

File: fs/vfs/fs_eventfd.c [hunk 3]
```diff
@@ -41,7 +41,7 @@
  ****************************************************************************/
 
 #ifndef CONFIG_EVENT_FD_VFS_PATH
-#define CONFIG_EVENT_FD_VFS_PATH "/dev"
+#define CONFIG_EVENT_FD_VFS_PATH "/dev/event"
```
Comment (pkarashchenko) [inline fs/vfs/fs_eventfd.c:44]:
```suggestion
#define CONFIG_EVENT_FD_VFS_PATH "/var/event"
```

Comment (xiaoxiang781216) [reply fs/vfs/fs_eventfd.c]: Done.

File: fs/vfs/fs_eventfd.c [hunk 4]
```diff
@@ -66,10 +66,10 @@ struct eventfd_priv_s
...
   eventfd_waiter_sem_t *rdsems; /* List of blocking readers */
   eventfd_waiter_sem_t *wrsems; /* List of blocking writers */
-  eventfd_t counter;            /* eventfd counter */
-  size_t    minor;              /* eventfd minor number */
-  uint8_t   crefs;              /* References counts on eventfd (max: 255) */
-  uint8_t   mode_semaphore;     /* eventfd mode (semaphore or counter) */
+  eventfd_t    counter;         /* eventfd counter */
+  unsigned int minor;           /* eventfd minor number */
+  uint8_t      crefs;           /* References counts on eventfd (max: 255) */
+  uint8_t      mode_semaphore;  /* eventfd mode (semaphore or counter) */
```
Comment (pkarashchenko) [inline fs/vfs/fs_eventfd.c:72]: Should we use `bool` here instead?

Comment (xiaoxiang781216) [reply fs/vfs/fs_eventfd.c]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**